crypto/random/rand_utils.go

```diff
@@ -49,3 +53,27 @@ func EvaluateDistributionUniformity(t *testing.T, distribution []float64) {
 	mean := stat.Mean(distribution, nil)
 	assert.Greater(t, tolerance*mean, stdev, fmt.Sprintf("basic randomness test failed: n: %d, stdev: %v, mean: %v", len(distribution), stdev, mean))
 }
+
+// computes a bijection from the set of all permutations
+// into the the set [0, n!-1] (where `n` is the size of input `perm`).
+// input `perm` is assumed to be a correct permutation of the set [0,n-1]
+// (not checked in this function).
+func EncodePermutation(perm []int) int {
+	r := make([]int, len(perm))
+	// generate Lehmer code
+	for i, x := range perm {
+		for _, y := range perm[i+1:] {
+			if y < x {
+				r[i]++
+			}
+		}
+	}
+	// Convert to an integer following the factorial number system
+	m := 0
```

Comment:
```suggestion
	// We construct `m` by taking the sum of all elements of `r`, each multiplied by the factorial of their index:
	// m = ∑ r(i) * i!, i ∈ [0,n-1]
	m := 0
```

Reply:
This comment is more explanatory but it doesn't explain why it makes sense to compute `m` (why there is a bijection between `[0,n-1] x [0,n-2] x ... x [0,1] x [0]` and `[0, n! -1] ). 
I suggest to add a link to the full explanation instead.

Reply:
https://github.com/onflow/flow-go/pull/4844/commits/03b9532779fe2e8c9504bc3472e9131dfc7d0c1f